server/src/main/java/org/elasticsearch/transport/TransportHandshaker.java

```diff
@@ -181,9 +183,17 @@ static final class HandshakeRequest extends TransportRequest {
             }
             if (remainingMessage == null) {
                 version = null;
+                minimumCompatibilityVersion = null;
             } else {
                 try (StreamInput messageStreamInput = remainingMessage.streamInput()) {
                     this.version = Version.readVersion(messageStreamInput);
+                    Version minimumCompatibilityVersion;
+                    try {
+                        minimumCompatibilityVersion = Version.readVersion(messageStreamInput);
```

Comment:
is this safe to do? (attempt to read more bytes from a < 5.10.1 version which could read the next bytes in the stream?)
wouldn't be safer to read the minCompatVersion only based on a check on the `version`?

Reply:
Yes this is safe to do as these messages are expected to have a variable length since https://github.com/elastic/elasticsearch/pull/36171 (it supports to a "future data" to the request). 

I've also verified this by manually running mixed versions.

Reply:
thx!